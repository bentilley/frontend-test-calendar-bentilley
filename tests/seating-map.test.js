import React from 'react';
import renderer from 'react-test-renderer';

import SeatingMap from 'components/seating/seating-map';

const seatingData = {
  'seats': {
    'seat': {
      x: 5,
      y: 5
    }
  },
  'text-elements' : [
    {
      x: 10,
      y: 10,
      content: "STAGE",
      size: 16
    }
  ],
  'shapes': [
    {
      x: 15,
      y: 15,
      width: 20,
      height: 20
    }
  ]
};

describe('Seating component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<SeatingMap />);
    const instance = component.getInstance();
    const result = instance.render();
    expect(result).toMatchSnapshot();
  });
  it('renderPlan calls subfuncs correctly', () => {
    const component = renderer.create(
      <SeatingMap
      />);
    const instance = component.getInstance();
    instance.renderSeat = jest.fn();
    instance.renderText = jest.fn();
    instance.renderShape = jest.fn();
    instance.renderPlan({}, seatingData);
    expect(instance.renderSeat).toBeCalledWith(expect.anything(), 5, 5, '#ccc');
    expect(instance.renderText).toBeCalledWith(expect.anything(), 10, 10, 'STAGE', 16, '#444');
    expect(instance.renderShape).toBeCalledWith(expect.anything(), 15, 15, 20, 20, '#444');
  });
});
