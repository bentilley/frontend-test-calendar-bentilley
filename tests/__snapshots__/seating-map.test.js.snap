// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seating component renders correctly 1`] = `
<div
  className="seating-map"
>
  <canvas
    height={500}
    width={500}
  />
</div>
`;
